service/controller/resource/instance/create_drain_old_worker_nodes.go

```diff
@@ -53,7 +53,13 @@ func (r *Resource) drainOldWorkerNodesTransition(ctx context.Context, obj interf
 
 	var nodesPendingDraining int
 	for _, i := range allWorkerInstances {
-		if *i.LatestModelApplied {
+		old, err := r.isWorkerInstanceFromPreviousRelease(ctx, cr, i)
+		if err != nil {
+			return DeploymentUninitialized, nil
```

Comment:
Just ignore error and return reconciliation back to the start when an error happens here? (just to check if it was intended)

Reply:
probably wrong and coming from a copy paste from another part. It was late guys :(

Reply:
but we can't postpone IMHO, let's merge.